### PR TITLE
Feat: add implementation of scanner.

### PR DESCRIPTION
**Pull request by MbIXjkee**

### CODE REVIEW

File: lib/src/scanner_manager.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  /// Due to information coming as events, we wait for the next event for some time.
+  /// This value is the default duration between events in the data sequence read once.
+  static const Duration _defaultDuration = Duration(milliseconds: 300);
+
+  /// The event source for devices that work as a hardware keyboard.
+  ///
+  /// See https://developer.android.com/reference/android/view/InputDevice#SOURCE_KEYBOARD
+  static const int _hardwareKeyboardSourceEvent = 257;
+
+  /// List of default target event sources, if special doesn't set
```
Comment (dtoton) [inline lib/src/scanner_manager.dart]: what do you mean by "if special doesn't set"?

File: lib/src/scanner_manager.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    this.targetSources = _defaultTargetEventSources,
+    KeyEventManager? keyEventManager,
+    Duration? eventDuration,
+  })  : _keyEventManager =
+            keyEventManager ?? ServicesBinding.instance.keyEventManager,
+        _duration = eventDuration ?? _defaultDuration {
+    _keyEventManager.keyMessageHandler = _handle;
+  }
+
+  /// Publisher that notify about reading info by scanner
```
Comment (dtoton) [inline lib/src/scanner_manager.dart]: should be: ...that 'notifies' about...

File: pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: qr_reader
+description: Implementation of the listener QR scanners that handle output as keyboard.
```
Comment (dtoton) [inline pubspec.yaml]: Implementation of the QR scanner listener that handles output as keyboard [events].

File: lib/src/scanner_manager.dart (hunk 1 above)
Comment (MbIXjkee) [reply lib/src/scanner_manager.dart]: If the user passes his own list to the constructor

**Review by dtoton — APPROVED** [review]

**Pull request merged**